Disallow abbreviated command-line options (#5469)
pytest-dev/pytest: changelog/1149.feature.rst

> Pytest no longer accepts prefixes of command-line arguments.

Reviewer: ```suggestion
Pytest no longer accepts abbreviation of command-line arguments, for example typing ``pytest --doctest-mod`` inplace of ``--doctest-modules``.
```

> Pytest no longer accepts prefixes of command-line arguments.

Reviewer: I think this should be a "removal" changelog, as we are dropping a "feature" (as much as it was considered a misfeature).

> This was previously allowed where the ``ArgumentParser`` thought it was unambigious,

Reviewer: ```suggestion

This was previously allowed where the ``ArgumentParser`` thought it was unambigious,

```

> because this could be incorrect due to delayed parsing of options for plugins.

Reviewer: ```suggestion
but this could be incorrect due to delayed parsing of options for plugins.
```

> See for example issues #1149, #3413, and #4009.

Reviewer: ```suggestion
See for example issues `#1149 <https://github.com/pytest-dev/pytest/issues/1149>`__, `#3413 <https://github.com/pytest-dev/pytest/issues/3413>`__, and `#4009 <https://github.com/pytest-dev/pytest/issues/4009>`__.
```
